### ISSUE TITLE
Fetching... зависает

### ISSUE THREAD

**Issue by andreypopov** [MaxXx1313/node-red-contrib-byfly#1]
Привет, не получаются данные, висит, есть идеи?

![2021-04-01_09-32-04](https://user-images.githubusercontent.com/1624839/113252939-30d2bb80-92cd-11eb-95ec-f119f4991da2.png)

1 Apr 09:30:51 - [info] [byfly-balance:1c2b2c89.030f43] TOKEN RECEIVED
1 Apr 09:30:51 - [info] [byfly-balance:1c2b2c89.030f43] byfly: connected
1 Apr 09:30:51 - [info] [byfly-balance:1c2b2c89.030f43] byfly: subscribe to topic

Требует капчу:)

**Comment by MaxXx1313** [comment]
Привет
Добавили капчу, да(

**Comment by andreypopov** [comment]
есть идеи?

**Comment by MaxXx1313** [comment]
да, можно попробовать пользователя ввести капчу и держать сессию, чтобы не закрывалась. Но не факт что через некоторе время снова не попросят капчу или оборвут сессию принудительно

**Comment by andreypopov** [comment]
Посмотрел сейчас новый логин, там смс приходит, я где-то месяц не заходил, сессия живая.

**Comment by andreypopov** [comment]
Ради интереса попробовал сделать,  у меня включена двухфакторная аутентификация

инструкция: нажать на инжект смс код, как придет код, вписать его в инжект и еще раз нажать, затем можно нажать на второй инжект

![2021-05-12_09-58-04](https://user-images.githubusercontent.com/1624839/117932221-c7e95500-b308-11eb-8554-fda6698dc724.png)

`[{"id":"341b8f5c.b39db","type":"http request","z":"e3944dc3.10f64","name":"авторизация","method":"POST","ret":"txt","paytoqs":"ignore","url":"https://myapi.beltelecom.by/api/v2/oauth/token","tls":"","persist":false,"proxy":"","authType":"","x":480,"y":220,"wires":[["72c4f2e7.d8318c"]]},{"id":"39b54b25.9d4c64","type":"debug","z":"e3944dc3.10f64","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":750,"y":300,"wires":[]},{"id":"337d7d0c.d73172","type":"inject","z":"e3944dc3.10f64","name":"СМС код","props":[{"p":"payload"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"num","x":120,"y":220,"wires":[["cb741d95.1b7ae"]]},{"id":"cb741d95.1b7ae","type":"function","z":"e3944dc3.10f64","name":"логин/пароль","func":"var code = msg.payload;\n\nmsg.payload = {\n    \"username\":\"\",\n    \"password\":\"\"\n};\nif (code) msg.payload.code = code;\n\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":300,"y":220,"wires":[["341b8f5c.b39db"]]},{"id":"25ab82a4.97b36e","type":"inject","z":"e3944dc3.10f64","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":75,"y":300,"wires":[["92ec459e.26d4c8"]],"l":false},{"id":"18f59517.70ee5b","type":"http request","z":"e3944dc3.10f64","name":"получить данные","method":"GET","ret":"txt","paytoqs":"ignore","url":"https://myapi.beltelecom.by/api/v2/contracts","tls":"","persist":false,"proxy":"","authType":"","x":490,"y":300,"wires":[["33c13fe7.00e87"]]},{"id":"72c4f2e7.d8318c","type":"json","z":"e3944dc3.10f64","name":"","property":"payload","action":"","pretty":false,"x":635,"y":220,"wires":[["b8b16138.15b5f"]],"l":false},{"id":"b8b16138.15b5f","type":"function","z":"e3944dc3.10f64","name":"сохранить токен","func":"if (\"access_token\" in msg.payload) {\n    flow.set(\"access_token\", msg.payload.access_token);\n}\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":770,"y":220,"wires":[[]]},{"id":"92ec459e.26d4c8","type":"function","z":"e3944dc3.10f64","name":"токен","func":"msg.headers = {\n    Authorization: \"Bearer \"+flow.get(\"access_token\")\n}\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":270,"y":300,"wires":[["18f59517.70ee5b"]]},{"id":"33c13fe7.00e87","type":"json","z":"e3944dc3.10f64","name":"","property":"payload","action":"","pretty":false,"x":635,"y":300,"wires":[["39b54b25.9d4c64"]],"l":false}]`

**Comment by MaxXx1313** [comment]
Обновил ноду, используем авторизацию (такой же как твой подход), которая держится 600 дней. Потом придется логинится заново(
